### PR TITLE
Fix golangci linters

### PR DESCRIPTION
**Pull request by tompsota**
The warnings from `gocyclo` and `funlen` linters are now resolved.

Signed-off-by: Tomas Psota <tpsota@redhat.com>

**Is this a BUG FIX or a FEATURE ?**:

> Uncomment only one, leave it on its own line:
>
> /kind bug
> /kind enhancement

**What this PR does / why we need it**:

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: controllers/handler/nodenetworkconfigurationpolicy_controller.go [hunk 1]
```diff
@@ -157,14 +149,39 @@ func (r *NodeNetworkConfigurationPolicyReconciler) Reconcile(_ context.Context,
...
+) error {
+	err := r.Client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("Policy is not found, removing previous enactment if any")
+			err = r.deleteEnactmentForPolicy(request.NamespacedName.Name)
+			return err
+		}
+		log.Error(err, "Error retrieving policy")
+		// Error reading the object - requeue the request.
```
Comment (phoracek) [inline controllers/handler/nodenetworkconfigurationpolicy_controller.go:168]: This comment is not relevant in the scope of the helper function - the re-queue happens on the caller side

File: cmd/handler/main.go [hunk 2]
```diff
@@ -100,6 +92,15 @@ func mainHandler() int {
 
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opt)))
 
+	if mainHandler() == 1 {
```
Comment (rhrazdil) [inline cmd/handler/main.go:95]: Could we use a constant?

File: controllers/handler/nodenetworkconfigurationpolicy_controller.go [hunk 3]
```diff
@@ -122,17 +122,9 @@ func (r *NodeNetworkConfigurationPolicyReconciler) Reconcile(_ context.Context,
 	_ = context.Background()
 	log := r.Log.WithValues("nodenetworkconfigurationpolicy", request.NamespacedName)
 
-	// Fetch the NodeNetworkConfigurationPolicy instance
 	instance := &nmstatev1.NodeNetworkConfigurationPolicy{}
-	err := r.Client.Get(context.TODO(), request.NamespacedName, instance)
+	err := r.fetchNodeNetworkConfigurationPolicyInstance(instance, &request, &log)
```
Comment (rhrazdil) [inline controllers/handler/nodenetworkconfigurationpolicy_controller.go:126]: I'm not sure about the motivation for the function.
Also, in go, it's common practice to not use `get/set/fetch/lookup` prefixes in function names.

File: controllers/handler/nodenetworkconfigurationpolicy_controller.go [hunk 4]
```diff
@@ -157,14 +149,39 @@ func (r *NodeNetworkConfigurationPolicyReconciler) Reconcile(_ context.Context,
...
+func (r *NodeNetworkConfigurationPolicyReconciler) fetchNodeNetworkConfigurationPolicyInstance(
+	instance *nmstatev1.NodeNetworkConfigurationPolicy,
+	request *ctrl.Request,
+	log *logr.Logger,
+) error {
+	err := r.Client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("Policy is not found, removing previous enactment if any")
+			err = r.deleteEnactmentForPolicy(request.NamespacedName.Name)
```
Comment (rhrazdil) [inline controllers/handler/nodenetworkconfigurationpolicy_controller.go:164]: I don't like that this function, that according to its name merely "fetches" a NNCP also deletes something else.

File: controllers/handler/nodenetworkconfigurationpolicy_controller.go [hunk 5]
```diff
@@ -157,14 +149,39 @@ func (r *NodeNetworkConfigurationPolicyReconciler) Reconcile(_ context.Context,
...
+			return err
+		}
+		log.Error(err, "Error retrieving policy")
+		// Error reading the object - requeue the request.
+		return err
+	}
+	return nil
+}
 
+func (r *NodeNetworkConfigurationPolicyReconciler) reconcileEnactmentConditions(
```
Comment (rhrazdil) [inline controllers/handler/nodenetworkconfigurationpolicy_controller.go:174]: IMO, it doesn't make sense to call this reconcile*
as in k8s , reconcile is a method of a reconciler, that is called by a controller.

File: pkg/policyconditions/conditions.go [hunk 6]
```diff
@@ -178,75 +177,9 @@ func update(apiWriter client.Client, apiReader client.Reader, policyReader clien
...
-		numberOfFinishedEnactments := enactmentsCountByCondition.Available() +
-			enactmentsCountByCondition.Failed() +
-			enactmentsCountByCondition.Aborted()
-
-		logger.Info(
-			fmt.Sprintf("numberOfNmstateMatchingNodes: %d, enactments count: %s",
-				numberOfNmstateMatchingNodes,
-				enactmentsCountByCondition),
-		)
+		setPolicyStatus(policy, &nmstateMatchingNodes, &enactments, &logger)
```
Comment (rhrazdil) [inline pkg/policyconditions/conditions.go:180]: I think that abstracting the status logic to a separate function could be a benefit, but can we limit the function to do only that?
And collect/calculate the node statuses outside of this function

**Review by rhrazdil — COMMENTED** [review]
The PR is pretty big, it would if it was spli into at least 2 commits, each delating with a different linter or a specific linter fix.

**Comment by tompsota** [comment]
> The PR is pretty big, it would if it was spli into at least 2 commits, each delating with a different linter or a specific linter fix.

Both linters were complaining about approximately the same issues, so we could only split it by files and not by linters, if that's ok

File: pkg/policyconditions/conditions.go [hunk 7]
```diff
@@ -258,6 +201,88 @@ func update(apiWriter client.Client, apiReader client.Reader, policyReader clien
 	})
 }
 
+func setPolicyStatus(
+	policy *nmstatev1.NodeNetworkConfigurationPolicy,
+	nodeStatus *nodeStatusInfo,
+	logger *logr.Logger,
+) {
+	logger.Info(
```
Comment (rhrazdil) [inline pkg/policyconditions/conditions.go:209]: I think we could simply log this information before this function is called. Then we don't need the logger argument.

**Review by rhrazdil — CHANGES_REQUESTED** [review]
We need to work a bit on the second commit, I have only minor nits in the third commit.

File: cmd/handler/main.go [hunk 8]
```diff
@@ -254,12 +160,145 @@ func mainHandler() int {
...
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create NodeNetworkConfigurationEnactment controller", "controller", "NMState")
+		return err
+	}
+
+	return nil
+}
+
+func createHealthyFile() error {
+	// Handler runs with host networking so opening ports is problematic
```
Comment (rhrazdil) [inline cmd/handler/main.go:228]: This comment at this scope doesn't make much sense, it should be before the function call.

**Review by rhrazdil — CHANGES_REQUESTED** [review]
One last nit, looking good otherwise!

**Review by rhrazdil — APPROVED** [review]
/lgtm
/approve

**Pull request merged**